pkg/client/vault/client.go

```diff
@@ -205,3 +294,56 @@ func compressZCAP(zcap *zcapld.Capability) (string, error) {
 
 	return base64.URLEncoding.EncodeToString(compressed.Bytes()), nil
 }
+
+func lastElm(s, sep string) string {
+	all := strings.Split(s, sep)
+
+	return all[len(all)-1]
+}
+
+func buildEDVURI(h, vid, did string) string {
+	return fmt.Sprintf("%s/encrypted-data-vaults/%s/documents/%s", h, vid, did)
+}
+
+func prepareRemoteJWEKey(keyURL string, kmsImpl kms.KeyManager) ([]byte, string, error) {
+	kid := lastElm(keyURL, "/")
+
+	pubKeyBytes, err := kmsImpl.ExportPubKeyBytes(kid)
+	if err != nil {
+		return nil, "", fmt.Errorf("failed to retrieve main public key bytes from remote KMS: %w", err)
+	}
+
+	return pubKeyBytes, kid, nil
+}
+
+func encryptContent(wKMS KeyManager, wCrypto crypto.Crypto, cek []byte) (string, error) {
```

Comment:
replace `cek` with `content` since we're encrypting content, not cek in this function